conans/client/tools/win.py

```diff
@@ -316,7 +314,15 @@ def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_
             continue
         try:
             name_var, value = line.split("=", 1)
-            if not only_diff or os.environ.get(name_var) != value:
+            if only_diff:
+                old_value = os.environ.get(name_var)
+                if old_value != value:
+                    if old_value is not None and old_value in value:
+                        new_value = value.replace(old_value, "")
+                        new_env[name_var] = new_value + "%{0}%".format(name_var)
```

Comment:
It won't be expanded in python. So any build helper or recipe using vcvars will get a wrong value here.